nav2_mppi_controller/src/critics/goal_angle_critic.cpp

```diff
@@ -35,14 +36,26 @@ void GoalAngleCritic::initialize()
 
 void GoalAngleCritic::score(CriticData & data)
 {
-  if (!enabled_ || !utils::withinPositionGoalTolerance(
-      threshold_to_consider_, data.state.pose.pose, data.goal))
+  if (!enabled_) {
+    return;
+  }
+
+  geometry_msgs::msg::Pose active_goal;
+  if (enforce_path_inversion_) {
+    active_goal = utils::getLastPathPose(data.path);
+  } else {
+    active_goal = data.goal;
+  }
+
+  if (!utils::withinPositionGoalTolerance(
+      threshold_to_consider_, data.state.pose.pose, active_goal))
   {
     return;
   }
 
-  const auto goal_idx = data.path.x.size() - 1;
-  const float goal_yaw = data.path.yaws(goal_idx);
+  tf2::Quaternion goal_orientation_q;
+  tf2::fromMsg(active_goal.orientation, goal_orientation_q);
+  double goal_yaw = tf2::getYaw(goal_orientation_q);
```

Comment:
I believe this works with just a geometry_msgs/Quaternion too https://github.com/ros-navigation/navigation2/blob/6e58fbe2888cf88d3040edb4b083b8dc1dfd0cd6/nav2_behaviors/plugins/spin.cpp#L86